hubverse_annotator/app.py

```diff
@@ -17,47 +17,64 @@
 import streamlit as st
 
 
-def create_forecast_chart(
-    hubverse_table: pl.DataFrame, reference_date: str
+def create_quantile_forecast_chart(
+    hubverse_table: pl.DataFrame,
+    value_col: str = "value",
 ) -> alt.Chart:
     """
     Uses a hubverse table (polars) and a reference date to
-    display forecast quantiles faceted by model.
+    display quantile forecasts faceted by model. The
+    output_type of the hubverse table must therefore be
+    'quantile'.
     """
-    df = hubverse_table.filter(
-        pl.col("output_type") == "quantile"
-    ).with_columns(
-        [
-            pl.col("output_type_id").cast(pl.Utf8).alias("quantile_str"),
-            pl.col("output_type_id").alias("quantile_num"),
-        ]
+    # get quantile values
+    quantiles = hubverse_table["output_type_id"].unique().sort().to_list()
+    # pivot columns
+    pivot_columns = [
+        c
+        for c in hubverse_table.columns
+        if c not in ["output_type_id", value_col]
+    ]
+    # filter to quantile only rows and ensure quantiles are str for pivot
+    # also, pivot to wide, so quantiles ids are columns
+    df_wide = (
+        hubverse_table.filter(pl.col("output_type") == "quantile")
+        .with_columns(
+            [
+                pl.col("output_type_id").cast(pl.Utf8),
+            ]
+        )
+        .pivot(
+            on="output_type_id",
+            index=pivot_columns,
+            values=value_col,
+        )
+        .with_columns(pl.col("0.5").alias("median"))
     )
-
-    df = df.with_columns(
-        [(1 - (pl.col("quantile_num") - 0.5).abs() * 2).alias("opacity")]
+    # create base Chart for altair errorbands
+    base = alt.Chart(df_wide).encode(
+        x=alt.X("target_end_date:T", title="Target End Date")
     )
-    unique_dates = (
-        df.select(pl.col("target_end_date").cast(pl.Utf8).unique().sort())
-        .to_series()
-        .to_list()
+    # create median line and CI bands
+    line = base.mark_line(
+        color="blue", strokeWidth=2, interpolate="monotone"
+    ).encode(y=alt.Y("median:Q", title=None))
+    band_01 = base.mark_errorband(opacity=0.2, interpolate="monotone").encode(
+        y=alt.Y("0.05:Q", title="Forecast Value"),
+        y2="0.95:Q",
+        color=alt.value("cyan"),
```

Comment:
Please use the default color if it looks okay or pick a less harsh color. In general, stay away from named colors like `"red"`, `"blue"`, `"green"` etc.

See, for example, this post about the default color palette in R https://blog.r-project.org/2019/11/21/a-new-palette-for-r/